src/CompletionProvider.php

```diff
@@ -134,30 +132,25 @@ public function provideCompletion(PhpDocument $doc, Position $pos): CompletionLi
             || $node instanceof Node\Expr\StaticPropertyFetch
             || $node instanceof Node\Expr\ClassConstFetch
         ) {
-            if (!is_string($node->name)) {
-                // If the name is an Error node, just filter by the class
+            // If the name is an Error node, just filter by the class
+            if ($node instanceof Node\Expr\MethodCall || $node instanceof Node\Expr\PropertyFetch) {
+                // For instances, resolve the variable type
+                $prefixes = DefinitionResolver::getFqnsFromType(
+                    $this->definitionResolver->resolveExpressionNodeToType($node->var)
+                );
+            } else {
+                $prefixes = [$node->class instanceof \PhpParser\Node\Name ? (string)$node->class : ''];
```

Comment:
PhpParser does this pattern a lot - defining an abstract base class like `PhpParser\Node\Expr` and then several implementations in a namespace like `PhpParser\Node\Expr\Variable`. This works fine in all other cases. Could you check the actual value of `$node->class` in your error case?